# Duke Feature

- Part 1. Combine md_242 and md_302: make md
- Part 2. Confirm that "label" and "feature"
- Part 3. Study feature
- Part 4. Correction: tdp and pdp
- Part 5. Concat 6 dfs: Output a corrected csv file


**Output files (.csv):** opv_export_masked_add (md)_corrected (tdp, pdp).csv

In [1]:
import pandas as pd

### Read "feature"

In [2]:
feature = pd.read_csv("opv_export_masked.csv")
print("The shape of feature is", feature.shape)
feature.head()

The shape of feature is (500, 429)


pdf_filename  \
0  1.2.276.0.75.2.5.80.25.3.190123133923759.93180...   
1  1.2.276.0.75.2.5.80.25.3.190123133924562.93180...   
2  1.2.276.0.75.2.5.80.25.3.190123134232237.93180...   
3  1.2.276.0.75.2.5.80.25.3.190123134234418.93180...   
4  1.2.276.0.75.2.5.80.25.3.190123134640676.93180...   

                                        opv_filename   maskedid aedob_shift  \
0  1.2.276.0.75.2.5.80.25.3.190123133923519.93180...  VIP093528  04/18/1959   
1  1.2.276.0.75.2.5.80.25.3.190123133924241.93180...  VIP093528  04/18/1959   
2  1.2.276.0.75.2.5.80.25.3.190123134230637.93180...  VIP059015  12/08/1941   
3  1.2.276.0.75.2.5.80.25.3.190123134233302.93180...  VIP059015  12/08/1941   
4  1.2.276.0.75.2.5.80.25.3.190123134640210.93180...  VIP099723  02/28/1949   

  aeexamdate_shift eye  trial_rx_sphere  trial_rx_cylinder  trial_rx_axis  \
0       06/02/2011   L             2.25               -3.0            176   
1       06/02/2011   R             3.00               -3.0             14   
2       03/28/1998   L             0.00                0.0              0   
3       03/28/1998   R             0.75                0.0              0   
4       09/30/2012   L             5.25               -1.5             70   

   pupil_diameter  ...  aeexamtime test_type  vatype  fovealresult  \
0             NaN  ...         NaN       NaN     NaN           NaN   
1             NaN  ...         NaN       NaN     NaN           NaN   
2             NaN  ...         NaN       NaN     NaN           NaN   
3             NaN  ...         NaN       NaN     NaN           NaN   
4             NaN  ...         NaN       NaN     NaN           NaN   

  clinicalnotes sfstatus  sf  sfprob  cpsd cpsdprob  
0           NaN      NaN NaN     NaN   NaN      NaN  
1           NaN      NaN NaN     NaN   NaN      NaN  
2           NaN      NaN NaN     NaN   NaN      NaN  
3           NaN      NaN NaN     NaN   NaN      NaN  
4           NaN      NaN NaN     NaN   NaN      NaN  

[5 rows x 429 columns]

# Part 1. Combine md_242 and md_302
### Create a new column, md

In [3]:
import numpy as np
md = [0] * feature.shape[0]

for i in range(feature.shape[0]):
    if feature["md_242"].isnull()[i] == False:
        md[i] = feature["md_242"][i]
    else:
        md[i] = feature["md_302"][i]
len(md)

500

In [4]:
feature.insert(32, "md", md)
print("The shape of feature is", feature.shape)
feature.head()

The shape of feature is (500, 430)


pdf_filename  \
0  1.2.276.0.75.2.5.80.25.3.190123133923759.93180...   
1  1.2.276.0.75.2.5.80.25.3.190123133924562.93180...   
2  1.2.276.0.75.2.5.80.25.3.190123134232237.93180...   
3  1.2.276.0.75.2.5.80.25.3.190123134234418.93180...   
4  1.2.276.0.75.2.5.80.25.3.190123134640676.93180...   

                                        opv_filename   maskedid aedob_shift  \
0  1.2.276.0.75.2.5.80.25.3.190123133923519.93180...  VIP093528  04/18/1959   
1  1.2.276.0.75.2.5.80.25.3.190123133924241.93180...  VIP093528  04/18/1959   
2  1.2.276.0.75.2.5.80.25.3.190123134230637.93180...  VIP059015  12/08/1941   
3  1.2.276.0.75.2.5.80.25.3.190123134233302.93180...  VIP059015  12/08/1941   
4  1.2.276.0.75.2.5.80.25.3.190123134640210.93180...  VIP099723  02/28/1949   

  aeexamdate_shift eye  trial_rx_sphere  trial_rx_cylinder  trial_rx_axis  \
0       06/02/2011   L             2.25               -3.0            176   
1       06/02/2011   R             3.00               -3.0             14   
2       03/28/1998   L             0.00                0.0              0   
3       03/28/1998   R             0.75                0.0              0   
4       09/30/2012   L             5.25               -1.5             70   

   pupil_diameter  ...  aeexamtime test_type  vatype  fovealresult  \
0             NaN  ...         NaN       NaN     NaN           NaN   
1             NaN  ...         NaN       NaN     NaN           NaN   
2             NaN  ...         NaN       NaN     NaN           NaN   
3             NaN  ...         NaN       NaN     NaN           NaN   
4             NaN  ...         NaN       NaN     NaN           NaN   

  clinicalnotes sfstatus  sf  sfprob  cpsd cpsdprob  
0           NaN      NaN NaN     NaN   NaN      NaN  
1           NaN      NaN NaN     NaN   NaN      NaN  
2           NaN      NaN NaN     NaN   NaN      NaN  
3           NaN      NaN NaN     NaN   NaN      NaN  
4           NaN      NaN NaN     NaN   NaN      NaN  

[5 rows x 430 columns]

In [5]:
print("Contains NaN?", feature["md_242"].isnull().values.any())
print("Number of NaN:", feature["md_242"].isnull().sum().sum())

Contains NaN? True
Number of NaN: 14


In [6]:
print("Contains NaN?", feature["md_302"].isnull().values.any())
print("Number of NaN:", feature["md_302"].isnull().sum().sum())

Contains NaN? True
Number of NaN: 486


In [7]:
print("Contains NaN?", feature["md"].isnull().values.any())
print("Number of NaN:", feature["md"].isnull().sum().sum())

Contains NaN? False
Number of NaN: 0


# Part 2. Confirm that "label" and "feature" 
- share the identical first columns ---- **pdf_filename**

### Read "label"

In [8]:
label = pd.read_csv('gradings_agreement_labeled.csv')
label2 = pd.read_csv('gradings_agreement_raw.csv')

### Compare "feature" and "label"

In [9]:
feature["pdf_filename"].compare(label["pdf_filename"])

Empty DataFrame
Columns: [self, other]
Index: []

In [10]:
feature["pdf_filename"].compare(label2["pdf_filename"])

Empty DataFrame
Columns: [self, other]
Index: []

# Part 3. Study feature

- we have 429 features in total.

0. **meta**: 39 + 1
1. **thr**: 76
2. **td**: 76
3. **tdp**: 76
4. **pd**: 76
5. **pdp**: 76
6. **other**: 10

In [11]:
feature.columns

Index(['pdf_filename', 'opv_filename', 'maskedid', 'aedob_shift',
       'aeexamdate_shift', 'eye', 'trial_rx_sphere', 'trial_rx_cylinder',
       'trial_rx_axis', 'pupil_diameter',
       ...
       'aeexamtime', 'test_type', 'vatype', 'fovealresult', 'clinicalnotes',
       'sfstatus', 'sf', 'sfprob', 'cpsd', 'cpsdprob'],
      dtype='object', length=430)

In [12]:
len(feature.columns)

430

## 0. meta df
- meta

In [13]:
meta = feature[feature.columns[:40]]
print("The shape of meta is", meta.shape)
meta.head()

The shape of meta is (500, 40)


pdf_filename  \
0  1.2.276.0.75.2.5.80.25.3.190123133923759.93180...   
1  1.2.276.0.75.2.5.80.25.3.190123133924562.93180...   
2  1.2.276.0.75.2.5.80.25.3.190123134232237.93180...   
3  1.2.276.0.75.2.5.80.25.3.190123134234418.93180...   
4  1.2.276.0.75.2.5.80.25.3.190123134640676.93180...   

                                        opv_filename   maskedid aedob_shift  \
0  1.2.276.0.75.2.5.80.25.3.190123133923519.93180...  VIP093528  04/18/1959   
1  1.2.276.0.75.2.5.80.25.3.190123133924241.93180...  VIP093528  04/18/1959   
2  1.2.276.0.75.2.5.80.25.3.190123134230637.93180...  VIP059015  12/08/1941   
3  1.2.276.0.75.2.5.80.25.3.190123134233302.93180...  VIP059015  12/08/1941   
4  1.2.276.0.75.2.5.80.25.3.190123134640210.93180...  VIP099723  02/28/1949   

  aeexamdate_shift eye  trial_rx_sphere  trial_rx_cylinder  trial_rx_axis  \
0       06/02/2011   L             2.25               -3.0            176   
1       06/02/2011   R             3.00               -3.0             14   
2       03/28/1998   L             0.00                0.0              0   
3       03/28/1998   R             0.75                0.0              0   
4       09/30/2012   L             5.25               -1.5             70   

   pupil_diameter  ...  md_302 md_242    md  mdprob   psd psdprob   vfi  \
0             NaN  ...     NaN  -2.20 -2.20    0.05  4.04   0.005  96.0   
1             NaN  ...     NaN  -1.72 -1.72    0.10  2.15   0.050  97.0   
2             NaN  ...     NaN  -1.77 -1.77     NaN  2.18     NaN  99.0   
3             NaN  ...     NaN  -1.51 -1.51     NaN  1.96     NaN  98.0   
4             NaN  ...     NaN  -3.76 -3.76    0.01  1.80   0.100  98.0   

                     ght  fovealthreshold fovealthresholdprob  
0  Outside Normal Limits             36.0                 NaN  
1  Outside Normal Limits             36.0                 NaN  
2   Within Normal Limits              NaN                 NaN  
3   Within Normal Limits              NaN                 NaN  
4  Outside Normal Limits             32.0                0.05  

[5 rows x 40 columns]

## 1. Threshold df
- thr

In [14]:
thr = feature[feature.columns[40:116]]
print("The shape of thr is", thr.shape)
thr.head()

The shape of thr is (500, 76)


n9_s27_thr  n3_s27_thr  t3_s27_thr  t9_s27_thr  n15_s21_thr  n9_s21_thr  \
0         NaN         NaN         NaN         NaN          NaN          28   
1         NaN         NaN         NaN         NaN          NaN          24   
2         NaN         NaN         NaN         NaN          NaN          29   
3         NaN         NaN         NaN         NaN          NaN          25   
4         NaN         NaN         NaN         NaN          NaN          26   

   n3_s21_thr  t3_s21_thr  t9_s21_thr  t15_s21_thr  ...  n15_i21_thr  \
0          28          26          29          NaN  ...          NaN   
1          26          28          24          NaN  ...          NaN   
2          27          24          26          NaN  ...          NaN   
3          25          24          24          NaN  ...          NaN   
4          22          23          23          NaN  ...          NaN   

   n9_i21_thr  n3_i21_thr  t3_i21_thr  t9_i21_thr  t15_i21_thr  n9_i27_thr  \
0          11          24          27          29          NaN         NaN   
1          26          26          25          28          NaN         NaN   
2          26          27          26          28          NaN         NaN   
3          28          27          26          26          NaN         NaN   
4          24          26          26          21          NaN         NaN   

   n3_i27_thr  t3_i27_thr  t9_i27_thr  
0         NaN         NaN         NaN  
1         NaN         NaN         NaN  
2         NaN         NaN         NaN  
3         NaN         NaN         NaN  
4         NaN         NaN         NaN  

[5 rows x 76 columns]

## 2. td df
- td

In [15]:
td = feature[feature.columns[116:192]]
print("The shape pf td is", td.shape)
td.head()

The shape pf td is (500, 76)


n9_s27_td  n3_s27_td  t3_s27_td  t9_s27_td  n15_s21_td  n9_s21_td  \
0        NaN        NaN        NaN        NaN         NaN          1   
1        NaN        NaN        NaN        NaN         NaN         -3   
2        NaN        NaN        NaN        NaN         NaN          3   
3        NaN        NaN        NaN        NaN         NaN         -1   
4        NaN        NaN        NaN        NaN         NaN          0   

   n3_s21_td  t3_s21_td  t9_s21_td  t15_s21_td  ...  n15_i21_td  n9_i21_td  \
0          0         -1          2         NaN  ...         NaN        -19   
1         -1          1         -3         NaN  ...         NaN         -3   
2          1         -2          1         NaN  ...         NaN         -2   
3         -1         -2         -1         NaN  ...         NaN          0   
4         -5         -3         -3         NaN  ...         NaN         -5   

   n3_i21_td  t3_i21_td  t9_i21_td  t15_i21_td  n9_i27_td  n3_i27_td  \
0         -6         -3         -1         NaN        NaN        NaN   
1         -3         -5         -2         NaN        NaN        NaN   
2         -1         -3         -1         NaN        NaN        NaN   
3         -1         -3         -3         NaN        NaN        NaN   
4         -4         -3         -8         NaN        NaN        NaN   

   t3_i27_td  t9_i27_td  
0        NaN        NaN  
1        NaN        NaN  
2        NaN        NaN  
3        NaN        NaN  
4        NaN        NaN  

[5 rows x 76 columns]

## 3. tdp df   ***
- tdp

In [16]:
tdp = feature[feature.columns[192:268]]
print("The shape of tdp is", tdp.shape)
tdp.head()

The shape of tdp is (500, 76)


n9_s27_tdp  n3_s27_tdp  t3_s27_tdp  t9_s27_tdp  n15_s21_tdp  n9_s21_tdp  \
0         NaN         NaN         NaN         NaN          NaN         NaN   
1         NaN         NaN         NaN         NaN          NaN         NaN   
2         NaN         NaN         NaN         NaN          NaN         NaN   
3         NaN         NaN         NaN         NaN          NaN         NaN   
4         NaN         NaN         NaN         NaN          NaN         NaN   

   n3_s21_tdp  t3_s21_tdp  t9_s21_tdp  t15_s21_tdp  ...  n15_i21_tdp  \
0         NaN         NaN         NaN          NaN  ...          NaN   
1         NaN         NaN         NaN          NaN  ...          NaN   
2         NaN         NaN         NaN          NaN  ...          NaN   
3         NaN         NaN         NaN          NaN  ...          NaN   
4         5.0         NaN         NaN          NaN  ...          NaN   

   n9_i21_tdp  n3_i21_tdp  t3_i21_tdp  t9_i21_tdp  t15_i21_tdp  n9_i27_tdp  \
0         0.5         1.0         5.0         NaN          NaN         NaN   
1         NaN         5.0         2.0         NaN          NaN         NaN   
2         NaN         NaN         NaN         NaN          NaN         NaN   
3         NaN         NaN         NaN         NaN          NaN         NaN   
4         5.0         5.0         5.0         1.0          NaN         NaN   

   n3_i27_tdp  t3_i27_tdp  t9_i27_tdp  
0         NaN         NaN         NaN  
1         NaN         NaN         NaN  
2         NaN         NaN         NaN  
3         NaN         NaN         NaN  
4         NaN         NaN         NaN  

[5 rows x 76 columns]

## 4. pd df
- pd_d

In [17]:
pd_d = feature[feature.columns[268:344]]
print("The shape of pd_d is", pd_d.shape)
pd_d.head()

The shape of pd_d is (500, 76)


n9_s27_pd  n3_s27_pd  t3_s27_pd  t9_s27_pd  n15_s21_pd  n9_s21_pd  \
0        NaN        NaN        NaN        NaN         NaN        0.0   
1        NaN        NaN        NaN        NaN         NaN       -4.0   
2        NaN        NaN        NaN        NaN         NaN        2.0   
3        NaN        NaN        NaN        NaN         NaN       -2.0   
4        NaN        NaN        NaN        NaN         NaN        2.0   

   n3_s21_pd  t3_s21_pd  t9_s21_pd  t15_s21_pd  ...  n15_i21_pd  n9_i21_pd  \
0        0.0       -2.0        1.0         NaN  ...         NaN      -19.0   
1       -2.0        1.0       -4.0         NaN  ...         NaN       -3.0   
2        0.0       -2.0        0.0         NaN  ...         NaN       -3.0   
3       -2.0       -2.0       -2.0         NaN  ...         NaN       -1.0   
4       -3.0       -1.0       -1.0         NaN  ...         NaN       -2.0   

   n3_i21_pd  t3_i21_pd  t9_i21_pd  t15_i21_pd  n9_i27_pd  n3_i27_pd  \
0       -6.0       -4.0       -1.0         NaN        NaN        NaN   
1       -4.0       -5.0       -2.0         NaN        NaN        NaN   
2       -2.0       -4.0       -2.0         NaN        NaN        NaN   
3       -2.0       -4.0       -4.0         NaN        NaN        NaN   
4       -2.0       -1.0       -6.0         NaN        NaN        NaN   

   t3_i27_pd  t9_i27_pd  
0        NaN        NaN  
1        NaN        NaN  
2        NaN        NaN  
3        NaN        NaN  
4        NaN        NaN  

[5 rows x 76 columns]

## 5. pdp df ***
- pdp

In [18]:
pdp = feature[feature.columns[344:420]]
print("The shape of pdp is", pdp.shape)
pdp.head()

The shape of pdp is (500, 76)


n9_s27_pdp  n3_s27_pdp  t3_s27_pdp  t9_s27_pdp  n15_s21_pdp  n9_s21_pdp  \
0         NaN         NaN         NaN         NaN          NaN         NaN   
1         NaN         NaN         NaN         NaN          NaN         NaN   
2         NaN         NaN         NaN         NaN          NaN         NaN   
3         NaN         NaN         NaN         NaN          NaN         NaN   
4         NaN         NaN         NaN         NaN          NaN         NaN   

   n3_s21_pdp  t3_s21_pdp  t9_s21_pdp  t15_s21_pdp  ...  n15_i21_pdp  \
0         NaN         NaN         NaN          NaN  ...          NaN   
1         NaN         NaN         NaN          NaN  ...          NaN   
2         NaN         NaN         NaN          NaN  ...          NaN   
3         NaN         NaN         NaN          NaN  ...          NaN   
4         NaN         NaN         NaN          NaN  ...          NaN   

   n9_i21_pdp  n3_i21_pdp  t3_i21_pdp  t9_i21_pdp  t15_i21_pdp  n9_i27_pdp  \
0         0.5         1.0         5.0         NaN          NaN         NaN   
1         NaN         5.0         2.0         NaN          NaN         NaN   
2         NaN         NaN         NaN         NaN          NaN         NaN   
3         NaN         NaN         NaN         NaN          NaN         NaN   
4         NaN         NaN         NaN         1.0          NaN         NaN   

   n3_i27_pdp  t3_i27_pdp  t9_i27_pdp  
0         NaN         NaN         NaN  
1         NaN         NaN         NaN  
2         NaN         NaN         NaN  
3         NaN         NaN         NaN  
4         NaN         NaN         NaN  

[5 rows x 76 columns]

## 6. tail

In [19]:
tail = feature[feature.columns[420:430]]
print("The shape of tail is", tail.shape)
tail.head()

The shape of tail is (500, 10)


aeexamtime  test_type  vatype  fovealresult  clinicalnotes  sfstatus  sf  \
0         NaN        NaN     NaN           NaN            NaN       NaN NaN   
1         NaN        NaN     NaN           NaN            NaN       NaN NaN   
2         NaN        NaN     NaN           NaN            NaN       NaN NaN   
3         NaN        NaN     NaN           NaN            NaN       NaN NaN   
4         NaN        NaN     NaN           NaN            NaN       NaN NaN   

   sfprob  cpsd  cpsdprob  
0     NaN   NaN       NaN  
1     NaN   NaN       NaN  
2     NaN   NaN       NaN  
3     NaN   NaN       NaN  
4     NaN   NaN       NaN

In [20]:
76 * 5 + (39 + 1) + 10

430

# Part 4. Correction: tdp and pdp

In [21]:
# 1. 
thr = feature[feature.columns[40:116]]
print("The shape of thr is", thr.shape)

The shape of thr is (500, 76)


In [22]:
# 2. 
td = feature[feature.columns[116:192]]
print("The shape pf td is", td.shape)

The shape pf td is (500, 76)


In [23]:
# 4. 
pd_d = feature[feature.columns[268:344]]
print("The shape of pd_d is", pd_d.shape)
type(pd_d)

The shape of pd_d is (500, 76)


pandas.core.frame.DataFrame

### 3. tdp

In [24]:
tdp = feature[feature.columns[192:268]]

tdp = tdp.replace(0.5, 4)
tdp = tdp.replace(1, 3)
####tdp = tdp.replace(2, 2)
tdp = tdp.replace(5, 1)

####

tdp.iloc[:, [5, 6, 7, 8, 
             11, 12, 13, 14, 15, 16, 
             19, 20, 21, 22, 23, 24, 25, 26, 
             28, 29, 30, 31, 32, 33, 34, 
             36, 
             38, 39, 40, 41, 42, 43, 44, 
             46, 
             49, 50, 51, 52, 53, 54, 55, 56, 
             59, 60, 61, 62, 63, 64, 
             67, 68, 69, 70]] = tdp.iloc[:, [5, 6, 7, 8, 
                                               11, 12, 13, 14, 15, 16, 
                                               19, 20, 21, 22, 23, 24, 25, 26, 
                                               28, 29, 30, 31, 32, 33, 34, 
                                               36, 
                                               38, 39, 40, 41, 42, 43, 44, 
                                               46, 
                                               49, 50, 51, 52, 53, 54, 55, 56, 
                                               59, 60, 61, 62, 63, 64, 
                                               67, 68, 69, 70]].fillna(0)

print("The shape of tdp is", tdp.shape)
tdp.head()

The shape of tdp is (500, 76)


n9_s27_tdp  n3_s27_tdp  t3_s27_tdp  t9_s27_tdp  n15_s21_tdp  n9_s21_tdp  \
0         NaN         NaN         NaN         NaN          NaN         0.0   
1         NaN         NaN         NaN         NaN          NaN         0.0   
2         NaN         NaN         NaN         NaN          NaN         0.0   
3         NaN         NaN         NaN         NaN          NaN         0.0   
4         NaN         NaN         NaN         NaN          NaN         0.0   

   n3_s21_tdp  t3_s21_tdp  t9_s21_tdp  t15_s21_tdp  ...  n15_i21_tdp  \
0         0.0         0.0         0.0          NaN  ...          NaN   
1         0.0         0.0         0.0          NaN  ...          NaN   
2         0.0         0.0         0.0          NaN  ...          NaN   
3         0.0         0.0         0.0          NaN  ...          NaN   
4         1.0         0.0         0.0          NaN  ...          NaN   

   n9_i21_tdp  n3_i21_tdp  t3_i21_tdp  t9_i21_tdp  t15_i21_tdp  n9_i27_tdp  \
0         4.0         3.0         1.0         0.0          NaN         NaN   
1         0.0         1.0         2.0         0.0          NaN         NaN   
2         0.0         0.0         0.0         0.0          NaN         NaN   
3         0.0         0.0         0.0         0.0          NaN         NaN   
4         1.0         1.0         1.0         3.0          NaN         NaN   

   n3_i27_tdp  t3_i27_tdp  t9_i27_tdp  
0         NaN         NaN         NaN  
1         NaN         NaN         NaN  
2         NaN         NaN         NaN  
3         NaN         NaN         NaN  
4         NaN         NaN         NaN  

[5 rows x 76 columns]

### 5. pdp

In [25]:
pdp = feature[feature.columns[344:420]]

pdp = pdp.replace(0.5, 4)
pdp = pdp.replace(1, 3)
####pdp = pdp.replace(2, 2)
pdp = pdp.replace(5, 1)

####

pdp.iloc[:, [5, 6, 7, 8, 
             11, 12, 13, 14, 15, 16, 
             19, 20, 21, 22, 23, 24, 25, 26, 
             28, 29, 30, 31, 32, 33, 34, 
             36, 
             38, 39, 40, 41, 42, 43, 44, 
             46, 
             49, 50, 51, 52, 53, 54, 55, 56, 
             59, 60, 61, 62, 63, 64, 
             67, 68, 69, 70]] = pdp.iloc[:, [5, 6, 7, 8, 
                                               11, 12, 13, 14, 15, 16, 
                                               19, 20, 21, 22, 23, 24, 25, 26, 
                                               28, 29, 30, 31, 32, 33, 34, 
                                               36, 
                                               38, 39, 40, 41, 42, 43, 44, 
                                               46, 
                                               49, 50, 51, 52, 53, 54, 55, 56, 
                                               59, 60, 61, 62, 63, 64, 
                                               67, 68, 69, 70]].fillna(0)
print("The shape of pdp is", pdp.shape)
pdp.head(5)

The shape of pdp is (500, 76)


n9_s27_pdp  n3_s27_pdp  t3_s27_pdp  t9_s27_pdp  n15_s21_pdp  n9_s21_pdp  \
0         NaN         NaN         NaN         NaN          NaN         0.0   
1         NaN         NaN         NaN         NaN          NaN         0.0   
2         NaN         NaN         NaN         NaN          NaN         0.0   
3         NaN         NaN         NaN         NaN          NaN         0.0   
4         NaN         NaN         NaN         NaN          NaN         0.0   

   n3_s21_pdp  t3_s21_pdp  t9_s21_pdp  t15_s21_pdp  ...  n15_i21_pdp  \
0         0.0         0.0         0.0          NaN  ...          NaN   
1         0.0         0.0         0.0          NaN  ...          NaN   
2         0.0         0.0         0.0          NaN  ...          NaN   
3         0.0         0.0         0.0          NaN  ...          NaN   
4         0.0         0.0         0.0          NaN  ...          NaN   

   n9_i21_pdp  n3_i21_pdp  t3_i21_pdp  t9_i21_pdp  t15_i21_pdp  n9_i27_pdp  \
0         4.0         3.0         1.0         0.0          NaN         NaN   
1         0.0         1.0         2.0         0.0          NaN         NaN   
2         0.0         0.0         0.0         0.0          NaN         NaN   
3         0.0         0.0         0.0         0.0          NaN         NaN   
4         0.0         0.0         0.0         3.0          NaN         NaN   

   n3_i27_pdp  t3_i27_pdp  t9_i27_pdp  
0         NaN         NaN         NaN  
1         NaN         NaN         NaN  
2         NaN         NaN         NaN  
3         NaN         NaN         NaN  
4         NaN         NaN         NaN  

[5 rows x 76 columns]

In [26]:
# 6. 
tail = feature[feature.columns[420:430]]
print("The shape of tail is", tail.shape)

The shape of tail is (500, 10)


In [27]:
meta.to_csv('/Users/yuhuicao/Desktop/1. Duke/0. meta.csv', index = False)
thr.to_csv('/Users/yuhuicao/Desktop/1. Duke/1. thr.csv', index = False)
td.to_csv('/Users/yuhuicao/Desktop/1. Duke/2. td.csv', index = False)
tdp.to_csv('/Users/yuhuicao/Desktop/1. Duke/3. tdp.csv', index = False)
pd_d.to_csv('/Users/yuhuicao/Desktop/1. Duke/4. pd.csv', index = False)
pdp.to_csv('/Users/yuhuicao/Desktop/1. Duke/5. pdp.csv', index = False)

# Columns Drop NaN

In [28]:
print("The shape before drop NaN is", thr.shape)
thr = thr.dropna(axis = 1)
print("The shape after drop NaN is", thr.shape)

The shape before drop NaN is (500, 76)
The shape after drop NaN is (500, 54)


In [29]:
print("The shape before drop NaN is", td.shape)
td = td.dropna(axis = 1)
print("The shape after drop NaN is", td.shape)

The shape before drop NaN is (500, 76)
The shape after drop NaN is (500, 54)


In [30]:
print("The shape before drop NaN is", tdp.shape)
tdp = tdp.dropna(axis = 1)
print("The shape after drop NaN is", tdp.shape)

The shape before drop NaN is (500, 76)
The shape after drop NaN is (500, 52)


In [31]:
print("The shape before drop NaN is", pd_d.shape)
pd_d = pd_d.dropna(axis = 1)
print("The shape after drop NaN is", pd_d.shape)

The shape before drop NaN is (500, 76)
The shape after drop NaN is (500, 0)


In [32]:
print("The shape before drop NaN is", pdp.shape)
pdp = pdp.dropna(axis = 1)
print("The shape after drop NaN is", pdp.shape)

The shape before drop NaN is (500, 76)
The shape after drop NaN is (500, 52)


# Part 5. Concat 6 dfs
- Output a corrected csv file

In [33]:
# thr, td, tdp, pd_d, pdp, tail
import pandas as pd
concatenated = pd.concat([meta, thr, td, tdp, pd_d, pdp, tail], axis=1)
print("The shape of concatenated is", concatenated.shape)
concatenated.head()

The shape of concatenated is (500, 262)


pdf_filename  \
0  1.2.276.0.75.2.5.80.25.3.190123133923759.93180...   
1  1.2.276.0.75.2.5.80.25.3.190123133924562.93180...   
2  1.2.276.0.75.2.5.80.25.3.190123134232237.93180...   
3  1.2.276.0.75.2.5.80.25.3.190123134234418.93180...   
4  1.2.276.0.75.2.5.80.25.3.190123134640676.93180...   

                                        opv_filename   maskedid aedob_shift  \
0  1.2.276.0.75.2.5.80.25.3.190123133923519.93180...  VIP093528  04/18/1959   
1  1.2.276.0.75.2.5.80.25.3.190123133924241.93180...  VIP093528  04/18/1959   
2  1.2.276.0.75.2.5.80.25.3.190123134230637.93180...  VIP059015  12/08/1941   
3  1.2.276.0.75.2.5.80.25.3.190123134233302.93180...  VIP059015  12/08/1941   
4  1.2.276.0.75.2.5.80.25.3.190123134640210.93180...  VIP099723  02/28/1949   

  aeexamdate_shift eye  trial_rx_sphere  trial_rx_cylinder  trial_rx_axis  \
0       06/02/2011   L             2.25               -3.0            176   
1       06/02/2011   R             3.00               -3.0             14   
2       03/28/1998   L             0.00                0.0              0   
3       03/28/1998   R             0.75                0.0              0   
4       09/30/2012   L             5.25               -1.5             70   

   pupil_diameter  ...  aeexamtime test_type  vatype  fovealresult  \
0             NaN  ...         NaN       NaN     NaN           NaN   
1             NaN  ...         NaN       NaN     NaN           NaN   
2             NaN  ...         NaN       NaN     NaN           NaN   
3             NaN  ...         NaN       NaN     NaN           NaN   
4             NaN  ...         NaN       NaN     NaN           NaN   

  clinicalnotes sfstatus  sf  sfprob  cpsd cpsdprob  
0           NaN      NaN NaN     NaN   NaN      NaN  
1           NaN      NaN NaN     NaN   NaN      NaN  
2           NaN      NaN NaN     NaN   NaN      NaN  
3           NaN      NaN NaN     NaN   NaN      NaN  
4           NaN      NaN NaN     NaN   NaN      NaN  

[5 rows x 262 columns]

In [34]:
concatenated.to_csv('/Users/yuhuicao/Desktop/1. Duke/opv_export_masked_add (md)_corrected (tdp, pdp).csv', index = False)